### PR TITLE
PR: Return types for CompiledObjects using docstrings.py

### PR DESCRIPTION
**Pull request by bcolsen**
This pull request in conjunction with PR: #796 for numpydoc returns in docstrings solves issue #372 for numpy complied objects. This makes completions functions numpy.array to return an numpy.ndarray object resulting in returns like this:

    >>> import jedi; jedi.Script('import numpy as np; np.array([1,2,3]).a').completions()
    [<Completion: all>,
     <Completion: any>,
     <Completion: argmax>,
     <Completion: argmin>,
     <Completion: argpartition>,
     <Completion: argsort>,
     <Completion: astype>]

I have not yet written a test since I need to import numpy to do such. A standard python library with complied objects and sphinx documentation would be a better test.

### CODE REVIEW

File: jedi/evaluate/compiled/__init__.py [hunk 1]
```diff
@@ -196,9 +203,12 @@ def name(self):
         return FakeName(name, self)
 
     def _execute_function(self, params):
+        from jedi.evaluate import docstrings
```
Comment (bcolsen) [inline jedi/evaluate/compiled/__init__.py]: This is necessary due to cyclical importing of CompliedObject in analysis.py which is import by docstrings.py through iteratable.py

**Comment by ccordoba12** [comment]
@bcolsen, could you update #796 and this one to work with the new 0.10.0?

**Comment by bcolsen** [comment]
@ccordoba12, I can look at getting the spyder jedi patch to work for both Jedi 0.9.0 and 0.10.0.

If needed, I could work on #796 but I'm not familiar with the googledocs stuff.  I'm also going to wait on this PR until #796 goes through.

**Comment by ccordoba12** [comment]
> @ccordoba12, I can look at getting the spyder jedi patch to work for both Jedi 0.9.0 and 0.10.0

Yes, that'd be great, thanks!

**Comment by bcolsen** [comment]
Closing this because I submitted PR: #868 that includes an updated version of this code.

**Pull request closed without merging**